adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/net45/WindowsFormsWebAuthenticationDialogBase.cs

```diff
@@ -217,14 +217,14 @@ protected virtual void WebBrowserNavigateErrorHandler(object sender, WebBrowserN
             this.OnNavigationCanceled(e.StatusCode);
         }
 
-        private bool CheckForClosingUrl(Uri url)
+        private bool CheckForClosingUrl(object sender, Uri url)
         {
             // Make change here
             bool canClose = false;
             if (url.Authority.Equals(this.desiredCallbackUri.Authority, StringComparison.OrdinalIgnoreCase) &&
                 url.AbsolutePath.Equals(this.desiredCallbackUri.AbsolutePath, StringComparison.OrdinalIgnoreCase))
             {
-                this.Result = new AuthorizationResult(AuthorizationStatus.Success, url.OriginalString);
+                this.Result = new AuthorizationResult(AuthorizationStatus.Success, GetAuthenticationCode(((CustomWebBrowser)sender).Document));
```

Comment:
>GetAuthenticationCode(((CustomWebBrowser)sender).Document) [](start = 83, length = 58)

potential breaking as far as I can see. would it be an idea to evaluate the sender type/value and do some error handling?